### PR TITLE
[Fix #390] Workaround for orphaned java process on windows

### LINKED ISSUE #390: Residual JVM After Quitting
Whenever I quit emacs or run nrepl-quit I get an orphaned JVM process remaining.

Attached should be an image showing the emacs process tree after running nrepl-jack-in.

I also tested with CIDER 20131018.1553 from a fresh .emacs.d with just cider/clojure-mode and had the same problem.

nrepl-jvm-bug

### PR DESCRIPTION
**Pull request by benedekfazekas**
after quitting the REPL. Use `interrupt-process` instead of `kill-process` if OS
is windows.

-----------------

Before submitting the PR make sure the following things have been done (and denote this
by checking the relevant checkboxes):

- [x] The commits are consistent with our [contribution guidelines][1]
- [ ] You've added tests (if possible) to cover your change(s)
- [x] All tests are passing (`make test`)
- [x] The new code is not generating bytecode or `M-x checkdoc` warnings
- [x] You've updated the changelog (if adding/changing user-visible functionality)
- [ ] You've updated the readme (if adding/changing user-visible functionality)
- [ ] You've updated the refcard (if you made changes to the commands listed there)

### CODE REVIEW

**Comment by jonpither** [comment]
This will be extremely useful for us poor developers working in a Windows environment.

File: nrepl-client.el [hunk 1]
```diff
@@ -616,6 +616,11 @@ If NO-ERROR is non-nil, show messages instead of throwing an error."
 
 
 ;;; Client: Process Handling
+(defun nrepl--kill-process (proc os)
```
Comment (bbatsov) [inline nrepl-client.el]: What's the point in passing a dynamic variable here? I'd just use `system-type` internally.

File: nrepl-client.el [hunk 2]
```diff
@@ -616,6 +616,11 @@ If NO-ERROR is non-nil, show messages instead of throwing an error."
...
+  "Kill PROC using the appropriate, OS specific way."
+  (if (memq os (list 'cygwin 'windows-nt))
```
Comment (bbatsov) [inline nrepl-client.el]: Why don't you simply quote this list?

File: nrepl-client.el [hunk 3]
```diff
@@ -616,6 +616,11 @@ If NO-ERROR is non-nil, show messages instead of throwing an error."
 
 
 ;;; Client: Process Handling
+(defun nrepl--kill-process (proc os)
+  "Kill PROC using the appropriate, OS specific way."
```
Comment (bbatsov) [inline nrepl-client.el]: I'd explain here the need for this, so it's clear it's a form of a workaround.

**Pull request merged**